src/DependencyInjection/DI/test/DependencyInjectionEventSourceTests.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using Microsoft.Extensions.DependencyInjection.Specification.Fakes;
+using Xunit;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    [CollectionDefinition(nameof(EventSourceTests), DisableParallelization = true)]
+    public class EventSourceTests : ICollectionFixture<EventSourceTests>
+    {
+    }
+
+    [Collection(nameof(EventSourceTests))]
+    public class DependencyInjectionEventSourceTests: IDisposable
+    {
+        private readonly TestEventListener _listener = new TestEventListener();
+
+        public DependencyInjectionEventSourceTests()
+        {
+            _listener.EnableEvents(DependencyInjectionEventSource.Instance, EventLevel.Verbose);
+        }
+
+        [Fact]
+        public void ExistsWithCorrectId()
+        {
+            var esType = typeof(DependencyInjectionEventSource);
+
+            Assert.NotNull(esType);
+
+            Assert.Equal("Microsoft-Extensions-DependencyInjection", EventSource.GetName(esType));
+            Assert.Equal(Guid.Parse("27837f46-1a43-573d-d30c-276de7d02192"), EventSource.GetGuid(esType));
+            Assert.NotEmpty(EventSource.GenerateManifest(esType, "assemblyPathToIncludeInManifest"));
+        }
+
+        [Fact]
+        public void EmitsCallSiteBuiltEvent()
+        {
+            // Arrange
+            var serviceCollection = new ServiceCollection();
+            serviceCollection.AddSingleton<FakeDisposeCallback>();
+            serviceCollection.AddTransient<IFakeOuterService, FakeDisposableCallbackOuterService>();
+            serviceCollection.AddSingleton<IFakeMultipleService, FakeDisposableCallbackInnerService>();
+            serviceCollection.AddScoped<IFakeMultipleService, FakeDisposableCallbackInnerService>();
```

Comment:
Could you put a couple other types of call-sites in? A Factory and a Constant in particular would be nice.

Reply:
Done